src/Psy/TabCompletion/Matchers/AbstractMatcher.php

```diff
@@ -0,0 +1,166 @@
+<?php
+
+namespace Psy\TabCompletion\Matchers;
+
+use Psy\Context;
+use Psy\ContextAware;
+
+/**
+ * Class AbstractMatcher
+ * @package Psy\TabCompletion\Matchers
+ */
+abstract class AbstractMatcher implements ContextAware
+{
+    /** Syntax types */
+    const CONSTANT_SYNTAX = '^[A-Z0-9-_]+$';
+    const VAR_SYNTAX = '^\$[a-zA-Z0-9-_]+$';
```

Comment:
if you are referring to this note : 
> Note: For our purposes here, a letter is a-z, A-Z, and the ASCII characters from 127 through 255 (0x7f-0xff).

It means the **content** of the variable or constant, not the name, the real regex is not like this, or you will end getting constant names as weird as "T�OT�", being all this chars in this contant inside the ranges defined in the note,but definitively are not a valid constant name.